src/node/rpc/member_frontend.h

```diff
@@ -20,8 +22,147 @@
 #include <set>
 #include <sstream>
 
+#include <openenclave/attestation/verifier.h>
+#if defined(INSIDE_ENCLAVE) && !defined(VIRTUAL_ENCLAVE)
+#  include <openenclave/enclave.h>
+#else
+#  include <openenclave/host_verify.h>
+#endif
+
 namespace ccf
 {
+  class MemberTsr : public lua::TxScriptRunner
+  {
+    void setup_environment(
+      lua::Interpreter& li,
+      const std::optional<Script>& env_script) const override
+    {
+      auto l = li.get_state();
+      lua_register(
+        l, "pem_to_der", lua_pem_to_der);
+      lua_register(
+        l, "verify_cert_and_get_claims", lua_verify_cert_and_get_claims);
+
+      TxScriptRunner::setup_environment(li, env_script);
+    }
+
+    static int lua_pem_to_der(lua_State* l)
+    {
+      std::string pem = get_var_string_from_args(l);
+      std::vector<uint8_t> der = tls::make_verifier(pem)->der_cert_data();
+      nlohmann::json json = der;
+      lua::push_raw(l, json);
+      return 1;
+    }
+
+#if 0
+    // Uses the new API: oe_verify_attestation_certificate_with_evidence.
+    // Requires OE 0.11.
+    // The quote is assumed to be an OE attestation (with oe_attestation_header_t header)
+    // stored at 1.2.840.113556.10.1.2 in the cert.
+
+    static oe_result_t oe_verify_attestation_certificate_with_evidence_cb(
+      oe_claim_t* claims, size_t claims_length, void* arg)
+    {
+      auto claims_map = (std::map<std::string,std::vector<uint8_t>>*)arg;
+      for (size_t i=0; i < claims_length; i++)
+      {
+        std::string claim_name(claims[i].name);
+        std::vector<uint8_t> claim_value(claims[i].value, claims[i].value + claims[i].value_size);
+        claims_map->emplace(std::move(claim_name), std::move(claim_value));
+      }
+      return OE_OK;
+    }
+
+    static int lua_verify_cert_and_get_claims(lua_State* l)
+    {
+      LOG_INFO_FMT("lua_verify_cert_and_get_claims");
+      nlohmann::json json = lua::check_get<nlohmann::json>(l, -1);
+      std::vector<uint8_t> cert_der = json;
+
+      std::map<std::string, std::vector<uint8_t>> claims;
+
+      oe_verifier_initialize();
+      oe_result_t res = oe_verify_attestation_certificate_with_evidence(
+        cert_der.data(),
+        cert_der.size(),
+        oe_verify_attestation_certificate_with_evidence_cb,
+        &claims);
+
+      if (res != OE_OK)
+      {
+        // TODO should this raise an exception?
+        throw std::runtime_error("certificate not valid");
+      }
+
+      lua_newtable(l);
+      const int table_idx = -2;
+
+      for (auto const& item : claims)
+      {
+        LOG_INFO_FMT("claim: {}", item.first);
+        // TODO not all claims should be hex-encoded
```

Comment:
How do we decide which should and which shouldn't?

Reply:
I don't think there's a way to do it automatically. Probably the right solution is to expose conversion functions like `buf_to_hex(v)`, `buf_to_int(v)`, `buf_to_str(v)` to the gov script and let the developer do it.

Reply:
I think it makes more sense to wait with this after we switch to JS. This disabled code path uses the new OE functions of 0.11, so it will take a while anyway until this gets in and probably requires some iteration.